src/Reflectify/Reflectify.cs

```diff
@@ -156,7 +157,7 @@ public static bool IsSameOrInherits(this Type actualType, Type expectedType)
     {
         return actualType == expectedType ||
                expectedType.IsAssignableFrom(actualType) ||
-               (actualType.BaseType?.IsGenericType is true && actualType.BaseType?.GetGenericTypeDefinition() == expectedType);
+               (actualType.BaseType is { IsGenericType: true } && actualType.BaseType.GetGenericTypeDefinition() == expectedType);
```

Comment:
When switching to pattern matching, [CA1508](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508) kicked in with
![image](https://github.com/user-attachments/assets/fdb28665-d6de-4084-bd83-dd73cbe6a6bf)